feat: JWT 및 시큐리티 세팅

kimhanshin4 opened the pull request:
두근두근 맞죠맞조 팀 프로젝트 첫 PR!
./global/security/ 의 JwtAuthenticationFilter에 '로그인'도 구현되어 있습니다
=> 회원가입 + 로그인 구현

kimhanshin4 committed:
feat: JWT 및 시큐리티 세팅

kimhanshin4 committed:
feat: 회원가입 구현

junxtar commented on src/main/java/com/sparta/springnewsfeed/domain/user/controller/UserController.java:
```diff
@@ -0,0 +1,33 @@
+        return ResponseEntity.ok(CommonCode.OK.getMessage());
+    }
+
+    //로그아웃
```
자신이 앞으로 할일은
 //TODO: 로그아웃 이렇게 작성 해보아요~

junxtar commented on src/main/java/com/sparta/springnewsfeed/domain/user/entity/User.java:
```diff
@@ -0,0 +1,28 @@
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+//    @Column(name = "user_id")
```
주석문 해제해도 좋을거 같습니다

junxtar commented on src/main/java/com/sparta/springnewsfeed/global/config/WebSecurityConfig.java:
```diff
@@ -0,0 +1,75 @@
+                .requestMatchers(PathRequest.toStaticResources().atCommonLocations())
+                .permitAll() // resources 접근 허용 설정
+                .requestMatchers("/").permitAll() // 메인 페이지 요청 허가
+                .requestMatchers("/api/v1/user/**").permitAll() // '/api/user/'로 시작하는 요청 모두 접근 허가
```
여기도 v1설정 빼시구 users로 변경하겠습니다

"/" -> "api/**" 로 변경하겠습니다

junxtar reviewed (changes requested):
리뷰 확인해주시구 수정하고 다시 pr 부탁드립니다

kimhanshin4 committed:
fix: controller/UserController - 주석문 //TODO로 변경. URL수정. |
entity/User - 주석문 해제 및 email필드 삭제. |
config/WebSecurityConfig - .permitAll부분 수정.

junxtar commented on src/main/java/com/sparta/springnewsfeed/domain/user/controller/UserController.java:
```diff
@@ -0,0 +1,33 @@
+        return ResponseEntity.ok(CommonCode.OK.getMessage());
+    }
+
+    //TODO:로그아웃
```
좋습니다~

junxtar replied:
메서드명 같은 경우는 현재 클래스가 User Controller 이기 때문에 User Signup -> Signup으로 변경해도 좋습니다

kimhanshin4 committed:
fix: Controller 메서드명 변경 및 Service의 암호 인코딩

kimhanshin4 committed:
fix: Controller 메서드명 재변경
Signup->signupㅠㅠ
